detection_rules/main.py

```diff
@@ -39,29 +39,51 @@ def root():
 @click.option('--rule-type', '-t', type=click.Choice(CurrentSchema.RULE_TYPES), help='Type of rule to create')
 def create_rule(path, config, required_only, rule_type):
     """Create a detection rule."""
-    config = load_dump(config) if config else {}
+    contents = load_rule_contents(config) or {}
     try:
-        return Rule.build(path, rule_type=rule_type, required_only=required_only, save=True, **config)
+        return Rule.build(path, rule_type=rule_type, required_only=required_only, save=True, **contents)
     finally:
         rule_loader.reset()
 
 
 @root.command('load-from-file')
 @click.argument('infile', type=click.Path(dir_okay=False, exists=True), nargs=-1, required=False)
 @click.option('--directory', '-d', type=click.Path(file_okay=False, exists=True), help='Load files from a directory')
-def load_from_file(infile, directory):
+@click.option('--skip-failures', '-s', is_flag=True,
+              help='Continue looping through rules when files fail to load (-d required)')
+@click.pass_context
+def load_from_file(ctx, infile, directory, skip_failures):
     """Load rules from file(s)."""
     if infile:
         for rule_file in infile:
             rule_path = os.path.join(RULES_DIR, os.path.basename(rule_file))
-            rule = Rule(rule_path, load_dump(rule_file))
-            rule.save(as_rule=True, verbose=True)
+            contents = load_rule_contents(rule_file) or {}
```

Comment:
the `or {}` seems weird here. either this function should fail with an exception or return an empty dict